examples/aggregations/aggregations.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/apache/arrow/go/v10/arrow"
+	"github.com/apache/arrow/go/v10/arrow/memory"
+	"github.com/polarsignals/frostdb"
+	"github.com/polarsignals/frostdb/dynparquet"
+	schemapb "github.com/polarsignals/frostdb/gen/proto/go/frostdb/schema/v1alpha1"
+	"github.com/polarsignals/frostdb/query"
+	"github.com/polarsignals/frostdb/query/logicalplan"
+)
+
+// This example demonstrates how to use FrostDB's to aggregate data
+func main() {
+
+	// Create a new column store
+	columnstore, _ := frostdb.New()
+	defer columnstore.Close()
+
+	// Open up a database in the column store
+	database, _ := columnstore.DB(context.Background(), "weather_db")
+
+	// Define our simple schema of labels and values
+	schema, _ := aggregationSchema()
+
+	// Create a table named simple in our database
+	table, _ := database.Table(
+		"snowfall_table",
+		frostdb.NewTableConfig(schema),
+	)
+
+	// Create values to insert into the database. We support a dynamic structure for city to
+	// accommodate cities in different regions
+	type WeatherRecord struct {
+		City     interface{}
+		Day      string
+		Snowfall int64
+	}
+
+	type CityInProvince struct {
+		Name     string
+		Province string
+	}
+
+	type CityInState struct {
+		Name  string
+		State string
+	}
+
+	montreal := CityInProvince{Name: "Montreal", Province: "Quebec"}
+	toronto := CityInProvince{Name: "Toronto", Province: "Ontario"}
+	minneapolis := CityInState{Name: "Minneapolis", State: "Minnesota"}
+
+	_, _ = table.Write(context.Background(),
+		WeatherRecord{Day: "Mon", Snowfall: 20, City: montreal},
+		WeatherRecord{Day: "Tue", Snowfall: 00, City: montreal},
+		WeatherRecord{Day: "Wed", Snowfall: 30, City: montreal},
+		WeatherRecord{Day: "Thu", Snowfall: 25, City: montreal},
+		WeatherRecord{Day: "Fri", Snowfall: 10, City: montreal},
+		WeatherRecord{Day: "Mon", Snowfall: 15, City: toronto},
+		WeatherRecord{Day: "Tue", Snowfall: 25, City: toronto},
+		WeatherRecord{Day: "Wed", Snowfall: 30, City: toronto},
+		WeatherRecord{Day: "Thu", Snowfall: 00, City: toronto},
+		WeatherRecord{Day: "Fri", Snowfall: 05, City: toronto},
+		WeatherRecord{Day: "Mon", Snowfall: 40, City: minneapolis},
+		WeatherRecord{Day: "Tue", Snowfall: 15, City: minneapolis},
+		WeatherRecord{Day: "Wed", Snowfall: 32, City: minneapolis},
+		WeatherRecord{Day: "Thu", Snowfall: 10, City: minneapolis},
+		WeatherRecord{Day: "Fri", Snowfall: 12, City: minneapolis},
+	)
+
+	// Create a new query engine to retrieve data
+	engine := query.NewEngine(memory.DefaultAllocator, database.TableProvider())
+
+	// Weekly snowfall statistics by city:
```

Comment:
When reading `Weekly` I thought you are aggregating per week. Now I see that the data is only for a week. Maybe remove the word to reflect the query 😄

Reply:
Ah thanks @teleivo -- I can see how it might be confusing. I have updated the wording :+1: